vms/platformvm/service.go

```diff
@@ -969,6 +970,91 @@ func (s *Service) GetCurrentValidators(_ *http.Request, args *GetCurrentValidato
 	return nil
 }
 
+type GetSubnetOnlyValidatorArgs struct {
+	ValidationID ids.ID `json:"validationID"`
+}
+
+type GetSubnetOnlyValidatorReply struct {
+	SubnetID ids.ID     `json:"subnetID"`
+	NodeID   ids.NodeID `json:"nodeID"`
+	// PublicKey is the compressed BLS public key of the validator
+	PublicKey             types.JSONByteSlice `json:"publicKey"`
+	RemainingBalanceOwner platformapi.Owner   `json:"remainingBalanceOwner"`
+	DeactivationOwner     platformapi.Owner   `json:"deactivationOwner"`
+	StartTime             avajson.Uint64      `json:"startTime"`
+	Weight                avajson.Uint64      `json:"weight"`
+	MinNonce              avajson.Uint64      `json:"minNonce"`
+	// Balance is the remaining amount of AVAX this SoV has to pay the
+	// continuous fee, according to the last accepted state.
+	Balance avajson.Uint64 `json:"balance"`
+	// Height is the height of the last accepted block
+	Height avajson.Uint64 `json:"height"`
+}
+
+// GetSubnetOnlyValidator returns the SoV if it exists
+func (s *Service) GetSubnetOnlyValidator(r *http.Request, args *GetSubnetOnlyValidatorArgs, reply *GetSubnetOnlyValidatorReply) error {
+	s.vm.ctx.Log.Debug("API called",
+		zap.String("service", "platform"),
+		zap.String("method", "getSubnetOnlyValidator"),
+	)
+
+	s.vm.ctx.Lock.Lock()
```

Comment:
nit: readlock

Reply:
I'm not going to make this change until we have cleaned up the locking structure of the platformvm. Every API in this file grabs a write lock because the locking model makes no sense currently, and this isn't a performance concern.